public/js/endpoints.js

```diff
@@ -61,4 +61,124 @@ $(document).ready(function() {
+        console.log('Installing CHIPSEC on selected endpoints:', selectedIds);
+    });
+
+    addNodeForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        const osType = document.getElementById('osType').value;
+        const hostId = document.getElementById('hostId').value;
+
+        console.log('Submitting form with data:', { osType, hostId }); // Add log here
+
+        // Send AJAX request to add the new host
+        $.ajax({
+            url: '/endpoints.js/addNode',
+            method: 'POST',
+            data: {
+                osType,
+                hostId
+            },
+            success: function(response) {
+                // Handle success (e.g., refresh the list of endpoints)
+                console.log('Host added successfully:', response);
+                location.reload();
+            },
+            error: function(error) {
+                // Handle error
+                console.error('Error adding host:', error);
+            }
+        });
+    });
+});
+
+document.getElementById('authMeth').addEventListener('change', function() {
+    var authMeth = this.value;
+    var credentialsSection = document.getElementById('credentials');
+    var privateKeySection = document.getElementById('privateKeySection');
+    if (authMeth === 'pass') {
+        credentialsSection.style.display = 'block';
+        privateKeySection.style.display = 'none';
+    } else if (authMeth === 'key') {
+        credentialsSection.style.display = 'none';
+        privateKeySection.style.display = 'block';
+    }
+});
+
+document.getElementById('defaultKeyCheckbox').addEventListener('change', function() {
+    var privateKeyInput = document.getElementById('privateKey');
+    if (this.checked) {
+        privateKeyInput.disabled = true;
+    } else {
+        privateKeyInput.disabled = false;
+    }
+});
+
+addNodeForm.addEventListener('submit', function(event) {
+    event.preventDefault();
+    var osType = document.getElementById('osType').value;
+    var hostId = document.getElementById('hostId').value;
+    var authMeth = document.getElementById('authMeth').value;
+    var data = { osType: osType, hostId: hostId};
+    var username = document.getElementById('username').value;
+    data.username = username;
+    if (authMeth === 'pass') {
+        var password = document.getElementById('password').value;
+        data.password = password;
+    } else if (authMeth === 'key') {
+        var useDefaultKey = document.getElementById('defaultKeyCheckbox').checked;
+        if (!useDefaultKey) {
+            var privateKey = document.getElementById('privateKey').value;
+            data.privateKey = privateKey;
+        }
+        else {
+            data.privateKey = "default";
+        }
+    }
+
+    // Here, make your API call with the `data` object
+    console.log('Form data:', data);
+
+    // Example API call (uncomment and modify as needed)
+    fetch('/api/agents/addNode/' + osType + "/" + hostId, {
```

Comment:
Why is this different than the AJAX endpoint? does AJAX call here?

Reply:
WDYM AJAX?

Reply:
Got it after Mazi's response, resolved.